src/CQRS/LeanCode.CQRS.AspNetCore/Registration/CQRSObjectsRegistrationSource.cs

```diff
@@ -2,25 +2,23 @@
 using LeanCode.Components;
 using LeanCode.Contracts;
 using LeanCode.CQRS.Execution;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace LeanCode.CQRS.AspNetCore.Registration;
 
 internal class CQRSObjectsRegistrationSource
 {
-    private readonly TypesCatalog contractsCatalog;
-    private readonly TypesCatalog handlersCatalog;
+    private readonly IServiceCollection services;
 
-    public IEnumerable<CQRSObjectMetadata> Objects { get; }
+    public IEnumerable<CQRSObjectMetadata> Objects { get; private set; }
```

Comment:
```suggestion
    private readonly HashSet<CQRSObjectMetadata> objects = new(CQRSObjectMetadataEqualityComparer.Instance);

    public IReadOnlySet<CQRSObjectMetadata> Objects => objects;
```

And then below just do
```csharp
if (objects.Add(metadata))
{
    services.AddCQRSHandler(metadata);
}
```

`AddCQRSHandler` that accepts a single `CQRSObjectMetadata` doesn't seem to exist yet so go ahead and add one (and make `AddCQRSHandlers` call it in a loop instead).